From a4e7527a725bf4d42d6beb8665f7fb577f3e1298 Mon Sep 17 00:00:00 2001
From: Dandawate Saket <dsaket@ti.com>
Date: Sat, 5 Mar 2011 00:36:15 +0530
Subject: OMAP4: DSI: Increase in DSI line buffer size

The DSI line buffer size was not utilized fully,
so it has been increased to the maximum.
For DSI1 it is now 1364 and 1023 for DSI2.

Change-Id: I3af8e5b9fc12240e65acd3e7c5aba90acf90bff8
Signed-off-by: Dandawate Saket <dsaket@ti.com>
---
 drivers/video/omap2/dss/dsi.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/video/omap2/dss/dsi.c b/drivers/video/omap2/dss/dsi.c
index ed0067f..fb43d54 100644
--- a/drivers/video/omap2/dss/dsi.c
+++ b/drivers/video/omap2/dss/dsi.c
@@ -3104,14 +3104,18 @@ static void dsi_update_screen_dispc(struct omap_dss_device *dssdev,
 	struct dsi_struct *p_dsi;
 	enum omap_dsi_index ix;
 	unsigned channel;
-	/* line buffer is 1024 x 24bits */
-	/* XXX: for some reason using full buffer size causes considerable TX
-	 * slowdown with update sizes that fill the whole buffer */
-	const unsigned line_buf_size = 1023 * 3;
+	unsigned line_buf_size;
 
 	ix = (dssdev->channel == OMAP_DSS_CHANNEL_LCD) ? DSI1 : DSI2;
 	p_dsi = (ix == DSI1) ? &dsi1 : &dsi2;
 
+	/* line buffer is 1365/1024 x 24bits */
+	/* XXX: for some reason using full buffer size causes considerable TX
+	 * slowdown with update sizes that fill the whole buffer so used total
+	 * size - 1. for DSI1, the max size is 1365 but for DSI2, it is 1024.
+	 */
+	line_buf_size = (ix == DSI1) ? (1364 * 3) : (1023 * 3);
+
 	channel = p_dsi->update_channel;
 
 	DSSDBG("dsi_update_screen_dispc(%d,%d %dx%d)\n",
-- 
1.6.0.4

